### PR TITLE
Add ruff as external

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Required for Python formatting with Ruff

### CODE REVIEW

**Comment by aandvalenzuela** [comment]
please test

File: ruff.spec [hunk 1]
```diff
@@ -0,0 +1,6 @@
+### RPM external ruff 0.5.6
+Source: https://github.com/astral-sh/ruff/releases/download/%{realversion}/ruff-x86_64-unknown-linux-gnu.tar.gz
+
+%install
+tar -zxvf ruff-x86_64-unknown-linux-gnu.tar.gz
```
Comment (smuzaffar) [inline ruff.spec]: please use `%setup` to unpack the source

**Comment by aandvalenzuela** [comment]
please test

File: ruff.spec (hunk 1 above)
Comment (smuzaffar) [reply ruff.spec]: I think all you need is to add the following
```
%prep
%setup -n ruff-x86_64-unknown-linux-gnu

```

File: ruff.spec [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### RPM external ruff 0.5.6
+Source: https://github.com/astral-sh/ruff/releases/download/%{realversion}/ruff-x86_64-unknown-linux-gnu.tar.gz
+
```
Comment (smuzaffar) [inline ruff.spec:3]: add an empty `%build` section

File: scram-tools.file/tools/ruff/ruff.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<tool name="ruff" version="@TOOL_VERSION@">
+  <lib name="ruff"/>
```
Comment (smuzaffar) [inline scram-tools.file/tools/ruff/ruff.xml]: why `<lib name...>` ? Do this tool have a library?

File: scram-tools.file/tools/ruff/ruff.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+  <runtime name="PATH" value="$RUFF_BASE/bin" type="path"/>
```
Comment (smuzaffar) [inline scram-tools.file/tools/ruff/ruff.xml:6]: what is `$RUFF_BASE` set?

Comment (smuzaffar) [reply scram-tools.file/tools/ruff/ruff.xml:6]: please check other tool files to see how to properly add a toolfile

Comment (smuzaffar) [reply scram-tools.file/tools/ruff/ruff.xml:6]: e.g. see `lcov.xml`

File: ruff.spec [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+%setup
+tar -zxvf ruff-x86_64-unknown-linux-gnu.tar.gz
```
Comment (smuzaffar) [inline ruff.spec]: you need something like
```
%prep
%setup -n ruff-x86_64-unknown-linux-gnu
```

Comment (aandvalenzuela) [reply ruff.spec]: And `%setup` would already untar and move to `%{i}/bin`?

Comment (smuzaffar) [reply ruff.spec]: no, it only unpack the source ( please read rpm docs :-) )

**Comment by aandvalenzuela** [comment]
please test

- Let's try

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Pull request merged**